libtransact/src/scheduler/serial/core.rs

```diff
@@ -310,11 +310,20 @@ impl SchedulerCore {
                     self.try_schedule_next()?;
                 }
                 Ok(CoreMessage::ExecutionResult(task_notification)) => {
+                    let current_txn_id = self.current_txn.as_ref().expect(
+                        "received execution result but no current transaction is executing",
+                    );
                     match task_notification {
-                        ExecutionTaskCompletionNotification::Valid(context_id, _transaction_id) => {
-                            let transaction_id = self.current_txn.take().expect(
-                                "received execution result but no current transaction is executing",
-                            );
+                        ExecutionTaskCompletionNotification::Valid(context_id, transaction_id) => {
+                            if &transaction_id != current_txn_id {
+                                warn!(
```

Comment:
This should at least be logged as error, since the result will get lost.  Not sure if we should panic here still or not, since we're not keeping track of out-of-order results.

Reply:
I agree with logging as an error; I'm not sure about panicking, because it could be caused by some non-fatal bug in the executor. But I don't have a strong opinion.

Reply:
I updated this to log as an error instead of a warning. Per our offline discussions, I created a task to add a new error callback for Schedulers (https://jira.hyperledger.org/browse/TX-50)